torch/_inductor/select_algorithm.py

```diff
@@ -1977,9 +1977,13 @@ def generate(  # type: ignore[override]
             expected_input_args,
         )
 
-        full_input_nodes = tuple(
+        # `kernel_input_nodes` are the actual inputs that will be passed to the kernel,
+        # so e.g. views of the same input are not included. `codegen_input_nodes`
+        # includes views of inputs to preserve the kernel semantics
```

Comment:
Maybe add a comment that these will be the strides used to infer read/writes in `TemplateBuffer.extract_read_writes` ?